carousel/carousel.js

```diff
@@ -0,0 +1,228 @@
+const carouselAutoplayIntervalMap = new Map();
+
+/*
+    CarouselOptions may include:
+        - allowCycle = Boolean. Default is true. Whether the carousel can loop over items or not.
+        - showDots = Boolean. Default is true. Whether the dots are visible.
+        - buttonStyle = String. Additional style to give each button.
+        - carouselStyle = String. Additional style to give the carousel as a whole.
+        - autoplayTimeInMs = Number. Default is 6000. The time for each carousel to autoplay in ms.
+ */
+function createCarousel(carouselName, carouselItems, carouselOptions) {
```

Comment:
Since you are already in the context of carousel, it is unnecessary to add the `carousel` prefix to all arguments.